app/controllers/comments_controller.rb

```diff
@@ -3,20 +3,15 @@ class CommentsController < ApplicationController
 
   def new
     @comment = Comment.new
-    @user = User.where(id: @comment.user_id)
+    @post = Post.find(params[:post_id])
   end
 
   def create
-    @comment = Comment.new(comment_params)
-    @comment.user_id = current_user.id
-    @comment.user = current_user
-    @comment.post = @post
-
-    if @comment.save
-      on_comment_create_success
-    else
-      on_comment_create_failure
-    end
+    @user = current_user
+    @post = Post.find(params[:post_id])
+    @comment = Comment.new(user: @user, post: @post, text: params[:comment][:text])
+    @comment.save
+    redirect_to user_post_path(@user, @post)
```

Comment:
[REQUIRED] Kindly redirect the users properly when it is successful and when it is not. You can do something like this: 
```
 if @comment.save
      redirect_to post_path(@comment.post)
    else
      @article = Post.find(params[:post_id])
      redirect_to post_path(@post)
    end

```